### DISCUSSION TITLE
[Feature request] Export plugin generate images

### DISCUSSION THREAD

**Discussion by fancsali** [zim-desktop-wiki/zim-desktop-wiki#2200] [Ideas]
# Description
When using plugins that generate an image from code (ex. https://github.com/jaap-karssenberg/zim-wiki/wiki/UML-Diagrams-Plugin), it'd be nice to have another option next to the 'Edit...' in the dropdown – namely a quick way of exporting the generated image, for sharing with others.

# Rationale
The reasoning behind this is quite simple: although ideally one would be able to just export and share the whole page or whole sections of the wiki, but not too many organisations are up to adopting a completley seemless open-source workflow. And even if they'd be up for it, the likelihood of convincing everyone to change to Zim (i.e. my favourite) is slim.

So, I'd like to just be able to drop a diagram into an email, or upload to our non-Zim wiki so others can see.

**Comment by introt** [comment]
Afaik currently all of the generator plugins output their images as attachments, so you can find the image file there, no need to export.

I recall seeing an open feature request for copying images to clipboard (regardless of whether they're generated, pasted ...) using the right-click menu. A workaround to achieve that functionality is to use the integrated web server and copy images using ones browser.

**Comment by fancsali** [comment]
Yes, it's just an attachment, so it's not impossible to find it. Just as the other suggestion with the clipboard,  it's about the ease of the workflow.

It'd be a a great feature, if one could copy/share attached inline images with "one click".
